### PR TITLE
Add 'Change color' option to mixer channels

### PR DESCRIPTION
**Pull request by v0idzz**
This enables the user to choose whatever color they want for a mixer channel. This is my first... bigger(?) PR, so don't be surprised if something's wrong. :stuck_out_tongue_closed_eyes:

### CODE REVIEW

**Comment by DouglasDGI** [comment]
I gave it a test, it seems to work fine for the most part.  I did notice one very minor bug where resetting the color of the selected bus to default doesn't update the color on the screen until you click on the fader or switch to another fader.

Also, how the colors are handled may need some discussion.  From what I can see, any unselected channels are combined with the color black, which can make the channel colors look ugly.  Maybe instead, we should make the default-colored channels the dark color, and then mix the selected channel with the color white?  Any opinions?  Maybe that would break some LMMS themes, I'm not sure.

Otherwise, I did not notice any bugs or problems.  :+1:

File: src/gui/widgets/FxLine.cpp [hunk 1]
```diff
@@ -245,6 +265,40 @@ void FxLine::contextMenuEvent( QContextMenuEvent * )
...
+}
+
+
+
+void FxLine::resetColor()
+{
+	FxChannel * targetChannel = Engine::fxMixer()->effectChannel( m_channelIndex );
+	targetChannel->m_useStyleColor = true;
+	m_useStyleColor = true;
+}
```
Comment (PhysSong) [inline src/gui/widgets/FxLine.cpp]: I think you should add `update();` at the end of this function to fix the bug discovered by @DouglasDGI.

**Comment by PhysSong** [comment]
The bug fixed. :+1:

**Comment by noahb01** [comment]
I agree with @DouglasDGI, we should lighten the selected channel rather than darkening all the unselected ones. Qt has a function `QColor::lighten()` that could be used for this. I don't think its a big deal if it breaks themes, its not a major change and it will look different but not necessarily bad.

**Comment by qnebra** [comment]
It colors "song editor" too, or only mixer?

**Comment by freshstudio** [comment]
> It colors "song editor" too, or only mixer?

It has no effect on the song editor.

I agree with @DouglasGDI and @noahb01. Pushing changes which make it work the opposite way to what I have implemented.

**Comment by PhysSong** [comment]
To remove the GUI dependency added in this PR, some extra changes are needed to loading/saving FX channel settings.

**Comment by russiankumar** [comment]
Didn't know this existed... made a PR #5589 which does this same thing.

**Comment by claell** [comment]
@russiankumar Too bad. Is there any chance this can get reviewed, tested and merged soon?

**Comment by russiankumar** [comment]
I kind of see no point, considering the newer one can already be merged without any rebasing hassle, while this one will have to wait. The newer one was already tested by some as well, seems to work as expected, so I’d just go with that one. Though I want a mod to confirm this, since technically I’m biased.

**Comment by claell** [comment]
Probably the best, especially since this seems not to be very active at the moment.

**Comment by Spekular** [comment]
Superceded by #5589, which is now merged.

**Pull request closed without merging**